### PR TITLE
[libsndfile] Update to 1.0.29

### PR DESCRIPTION
**Pull request by evpobr**
Updates:

* Bump to official 1.0.29 release
* Two Git patches removed as merged to upstream
* Request Python3 when building shared library to generate export symbol files
* `CONTROL` replaced by `vcpkg.json`
* Removed `BUILD_EXECUTABLES` code. It was always `OFF` anyway
* Use new options instead of deprecated ones

Important note: target name is changed to`SndFile::sndfile` from upstream.

### CODE REVIEW

**Comment by evpobr** [comment]
Don't understand why Python is not found.

**Comment by evpobr** [comment]
Thanks @LilyWangL :smile:

**Comment by evpobr** [comment]
@LilyWangL , this PR is ready.

**Review by LilyWangL — APPROVED** [review]

File: ports/libsndfile/portfile.cmake [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
+    vcpkg_find_acquire_program(PYTHON3)
+endif()
 
 if("external-libs" IN_LIST FEATURES)
-    set(SNDFILE_WITHOUT_EXTERNAL_LIBS OFF)
+    set(SNDFILE_WITH_EXTERNAL_LIBS ON)
 else()
-    set(SNDFILE_WITHOUT_EXTERNAL_LIBS ON)
+    set(SNDFILE_WITH_EXTERNAL_LIBS OFF)
 endif()
```
Comment (BillyONeal) [inline ports/libsndfile/portfile.cmake:20]: It seems like this should do:

vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES external-libs SNDFILE_WITH_EXTERNAL_LIBS);

(and then below for `vcpkg_configure_cmake` do `${FEATURE_OPTIONS}`)

**Review by BillyONeal — APPROVED** [review]

**Review by evpobr — COMMENTED** [review]
@BillyONeal , it is final 1.0.29 release.

**Comment by BillyONeal** [comment]
@evpobr: Yes, but the old one was marked 1.0.29-10 and we want the version number to be increasing over time (assuming the upstream is going forward over time)

**Comment by BillyONeal** [comment]
(Note that our `version-portversion` convention in the CONTROL files was *not* semver)

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**